src/backend/plaf.rs

```diff
@@ -121,6 +122,39 @@ impl<F: PrimeField<Repr = [u8; 32]>, TraceArgs, StepArgs: Clone>
             plaf.lookups.push(plaf_lookup);
         }
 
+        if !self.circuit.exposed.is_empty() {
+            // Public column not pulled from Chiquito ir, because it's not stored anywhere.
+            // Therefore, we create a Plaf public column from scratch.
+            let plaf_public = ColumnPublic::new(String::from(
+                "exposed forward signal values in first step instance",
+            ));
+            plaf.columns.public.push(plaf_public);
+        }
+
+        for (index, (c_column, rotation)) in self.circuit.exposed.iter().enumerate() {
+            let public_column = pColumn {
+                kind: ColumnKind::Public,
+                index: 0, // Chiquito only has one public column, so the index is always 0.
+            };
+
+            let witness_index = self
+                .c_column_id_to_p_column_index
+                .get(&c_column.uuid())
+                .unwrap();
+
+            let witness_column = pColumn {
```

Comment:
@qwang98 shouldn't we retrieve the actual column from plaf, instead of making a new one?

Reply:
As seen in the prior comment. The actual column stored in `Plaf` struct is different from `Column` used to express Plaf's copy constraints. The copy constraints also take ownership of the `Column` struct, and that's why I make a new one. Again, the `Column` struct here is really just used to express copy constraints and therefore duplicating them should be fine.

```
pub struct CopyC {
    pub columns: (expr::Column, expr::Column),
    pub offsets: Vec<(usize, usize)>,
}

pub struct Column {
    pub kind: ColumnKind,
    pub index: usize,
}
```